react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+
+export default class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentIndex: 0,
+      intervalID: null
+    };
+    this.showNextIndex = this.showNextIndex.bind(this);
+    this.showPreviousIndex = this.showPreviousIndex.bind(this);
+    this.showClicked = this.showClicked.bind(this);
+  }
+
+  componentDidMount() {
+    const intervalID = setInterval(this.showNextIndex, 1000);
+    this.setState({ intervalID });
+  }
+
+  showNextIndex() {
+    this.state.currentIndex === 4
+      ? this.setState({ currentIndex: 0 })
+      : this.setState({ currentIndex: this.state.currentIndex + 1 });
+    this.controlInterval();
+  }
+
+  showPreviousIndex() {
+    this.state.currentIndex === 0
+      ? this.setState({ currentIndex: 4 })
+      : this.setState({ currentIndex: this.state.currentIndex - 1 });
+    this.controlInterval();
+  }
+
+  selectImage() {
+    const currentIndex = this.state.currentIndex;
+    return currentIndex === 0
+      ? this.props.pokemon.bulbasaur
+      : currentIndex === 1
+        ? this.props.pokemon.charmander
+        : currentIndex === 2
+          ? this.props.pokemon.squirtle
+          : currentIndex === 3
+            ? this.props.pokemon.pikachu
+            : currentIndex === 4 &&
+              this.props.pokemon.jigglypuff;
```

Comment:
The problem with this approach is that if the component is going to receive more / different pokemon through props, we have to update this method. Ultimately, our component should be generic enough that it doesn't really care if the input (props) are different.